spiceaidocs/docs/components/data-connectors/sharepoint.md

```diff
@@ -0,0 +1,62 @@
+---
+title: 'Sharepoint Data Connector'
+sidebar_label: 'Sharepoint Data Connector'
+description: 'Sharepoint Data Connector Documentation'
+pagination_prev: null
+---
+
+The Sharepoint Data Connector enables federated SQL queries on documents stored in Sharepoint.
+
+```yaml
+datasets:
+  - from: sharepoint:drive:Documents/path:/nuclear_secrets/
+    name: important_documents
+    params:
+      sharepoint_client_id: ${secrets:SPICE_SHAREPOINT_CLIENT_ID}
+      sharepoint_tenant_id: ${secrets:SPICE_SHAREPOINT_TENANT_ID}
+      sharepoint_client_secret: ${secrets:SPICE_SHAREPOINT_CLIENT_SECRET}
+```
+
+:::warning[Limitations]
+The sharepoint connector does not support creating a dataset from a single file (e.g. an Excel spreadsheet). Datasets must be created from a folder of documents.
+:::
+
+## Configuration
+### Parameters
+
+- `sharepoint_client_id`: Required. The client ID of the Azure AD (Entra) application.
+- `sharepoint_tenant_id`: Required. The tenant ID of the Azure AD (Entra) application.
+- `sharepoint_client_secret`: Optional. For service principal authentication. The client secret of the Azure AD (Entra) application.
+- `sharepoint_authorization_code`: Optional. For user authentication. The authorization code obtained from the OAuth2 flow (see `spice login sharepoint` [docs](cli/reference/login)).
+
+Note: Only one of `sharepoint_client_secret` or `sharepoint_authorization_code` is allowed.
+
+### `from` formats
+
+The `from` field in a Sharepoint dataset takes the following format:
+```yaml
+from: 'sharepoint:<drive_type>:<drive_id>/<subpath_type>:<subpath_value>'
+```
```

Comment:
Missing `file_format` and description about that?

Reply:
Because we only support documents, we don't inherently need `file_format`. We could use it, and apply it as a filter to files in a folder